src/components/Query.tsx

```diff
@@ -84,8 +87,10 @@ class QueryHandler extends Component<QueryHandlerProps, QueryHandlerState> {
   }
 }
 
-export type QueryProps = Omit<QueryHandlerProps, 'client'>;
+export type QueryProps<T> = Omit<QueryHandlerProps<T>, 'client'>;
 
-export const Query: FC<QueryProps> = props => (
-  <Consumer>{client => <QueryHandler {...props} client={client} />}</Consumer>
-);
+export function Query<T = any>(props: QueryProps<T>) {
```

Comment:
Here's where the default `any` comes in. I elected to refactor from `const` to `function` because of the strange syntax involved in supplying generics to `const` declarations with JSX, i.e. `export const Query = <T extends {}>(props: QueryProps<T>) = {}`. The above feels more readable to me.